packages/react-reconciler/src/ReactFiberCommitWork.new.js

```diff
@@ -892,7 +879,22 @@ function commitUnmount(
             const {destroy, tag} = effect;
             if (destroy !== undefined) {
               if ((tag & HookPassive) !== NoHookEffect) {
-                enqueuePendingPassiveHookEffectUnmount(current, effect);
+                effect.tag |= HookHasEffect;
+
+                current.effectTag |= Passive;
+
+                if (__DEV__) {
+                  // This flag is used to avoid warning about an update to an unmounted component
+                  // if the component has a passive unmount scheduled.
+                  // Presumably the listener would be cleaned up by that unmount.
+                  current.effectTag |= PassiveUnmountPendingDev;
```

Comment:
I think this could go on the deleted fiber only. In the warning function, can search up return path.

(Non-blocking)

Reply:
Hm, maybe...

If we weren't careful, we might miss legit warnings that way. We wouldn't want to incorrectly suppress a warning for a state update for a component that's within a tree that happened to have a subtree tag, even if the component in question doesn't have a passive effect scheduled. I guess we could walk the return path to see if an ancestor has `PassiveUnmountPendingDev`, then walk the Fiber receiving the update to see if its effect list contains a passive effect.

But as I write that, I realize that if we are willing to walk the effect list, we don't really need to look at an ancestor. We could just check for a `Passive` effect on the fiber an update is being scheduled for, then walk its effects list to confirm there's actually a destroy function. So I think we could maybe remove the `PassiveUnmountPendingDev` bit entirely?

Let me take a pass at that.

Reply:
So essentially this:
```js
// If there are pending passive effects unmounts for this Fiber,
// we can assume that they would have prevented this update.
if ((fiber.effectTag & PassiveUnmountPendingDev) !== NoEffect) {
  return;
}
```

Would become this:
```js
const alternate = fiber.alternate;
if ((fiber.effectTag & PassiveStatic) !== NoEffect) {
  const updateQueue: FunctionComponentUpdateQueue | null = (fiber.updateQueue: any);
  if (updateQueue !== null) {
    const lastEffect = updateQueue.lastEffect;
    if (lastEffect !== null) {
      const firstEffect = lastEffect.next;

      let effect = firstEffect;
      do {
        if (effect.destroy !== undefined) {
          if ((effect.tag & HookPassive) !== NoHookEffect) {
            return;
          }
        }
        effect = effect.next;
      } while (effect !== firstEffect);
    }
  }
}
```

The latter is clearly worse, but it's lazy *and* it's DEV only code, so it seems like an improvement.

Reply:
I don't know. I'll make this change but I don't know that it's worth doing. :)

Reply:
Cool, yeah this isn't super important either way. I was mostly thinking of what we'd have to do in the future to avoid visiting passive effect fibers during the layout phase.